### ISSUE TITLE
how to compute the bi-stochastic matrix

### ISSUE THREAD

**Issue by venuszhou** [yannadani/dlgm#2]
Hello, In the **main.py/class VGG_graph_matching/biStochastic_forward()** function:
```
S = v.view(n,m)
for i in range(N):
        S = torch.mm(S, torch.mm(torch.ones(1,n),S).inverse())
        S = torch.mm(torch.mv(S, torch.ones(m,1)).inverse(), S)
return S
```
The **A.inverse()** function need the matrix A to be a square matrix, so this cannot work?

**Comment by ninodimontalcino** [comment]
Hi,

since we focused on optical flow generation, we had no need for the "biStochastic" function. As you can also notice from the missing batch-wise implementation, this function is not completely implemented which is unfortunately not marked in the code. But we would be happy, if you would complete this missing function and add it to our re-implementation.

Cheers, Nino

**Comment by venuszhou** [comment]
@ninodimontalcino hey, thanks. I am also curious about that when you experiment on the Sintel dataset, how does the loss change? will it decrease significantly?
